Sources/TestsCommon/CommonTestUtils.swift

```diff
@@ -122,6 +122,73 @@ open class MongoSwiftTestCase: XCTestCase {
     }
 }
 
+/// Enumerates the different topology configurations that are used throughout the tests
+public enum TestTopologyConfiguration: String, Decodable {
+    case sharded
+    case replicaSet = "replicaset"
+    case single
+
+    /// Determines the topologyType of a client based on the reply returned by running an isMaster command
+    public init(isMasterReply: BSONDocument) throws {
+        // Check for symptoms of different topologies
+        if isMasterReply["msg"] != "isdbgrid" &&
+            isMasterReply["setName"] == nil &&
+            isMasterReply["isreplicaset"] != true {
+            self = .single
+        } else if isMasterReply["msg"] == "isdbgrid" {
+            self = .sharded
+        } else if isMasterReply["ismaster"] == true && isMasterReply["setName"] != nil {
+            self = .replicaSet
+        } else {
+            fatalError("Invalid test topology configuration")
```

Comment:
it might be helpful to include `isMasterReply` in the error string here, so we can use it to diagnose what went wrong if we ever hit this case.